### PR TITLE
feat: add more mappings for ibis ops and substrait scalar functions

### PR DESCRIPTION
**Pull request by richtia**
PR to add more mappings for ibis ops <-> substrait functions.

### CODE REVIEW

File: ibis_substrait/compiler/mapping.py [hunk 1]
```diff
@@ -1,33 +1,76 @@
...
     "Min": "min",
+    "Modulus": "modulus",
     "Multiply": "multiply",
+    "Negate": "negate",
     "Not": "not",
     "NotEquals": "not_equal",
+    "NotNull": "is_not_null",
+    "NullIf": "nullif",
     "Or": "or",
-    "StringLength": "string_length",  # don't know
```
Comment (richtia) [inline ibis_substrait/compiler/mapping.py:26]: added below as "char_length"

**Comment by richtia** [comment]
@gforsyth could you take a look at this and let me know if there are any other places that I need to make changes?

**Comment by gforsyth** [comment]
Hey @richtia -- this look good for the moment.  I'm currently adding in support for proper extension URI handling, so once I have that working we can keep building this on top of it.

**Comment by gforsyth** [comment]
thanks @richtia !

**Pull request merged**